boot/bootutil/src/bootutil_misc.c

```diff
@@ -837,3 +837,28 @@ boot_set_confirmed(void)
     flash_area_close(fap);
     return rc;
 }
+
+/**
+ * Populates the version information of the
+ * currently installed primary application
+ *
+ * @param[in] version Destination version structure buffer
+ * @return 0 on success; nonzero on failure.
+ */
+int boot_get_current_version(struct image_version *version)
+{
+    struct boot_loader_state state; // Dummy state
+    struct boot_status status;      // Dummy status
+    struct image_header hdr;
+    int rc;
+
+    rc = boot_read_image_header(&state, 0, &hdr, &status);
+    if(rc != 0) {
+        return rc;
+    }
+
+    // Copy the header's version struct over into the caller-supplied buffer
+    memcpy(version, &hdr.ih_ver, sizeof(struct image_version));
```

Comment:
What if someone passed in `version` equal to `NULL`?

Reply:
There is an assert, which suggests the API is not intended to cover this case. If assertions aren't tested, this will provoke undefined behavior.